### DISCUSSION TITLE
Grain boundary (bnds) problem

### DISCUSSION THREAD

**Discussion by mdh23rct** [idaholab/moose#25984] [Q&A Modules: Phase field]
Dear MOOSE experts,
Hope you are doing well! I am doing a two-step grain growth and recrystallization simulation using the phase field module of MOOSE.
1. In the first step, I am simulating grain growth to generate a polycrystal.
2. In the second step, I use the output file of grain growth model to compute recrystallization that nucleates new grains at abrupt locations of the polycrystal.

During recrystallization, I am finding new grains being added to the reserved OP (gr10). However, I am not seeing any change in the grain boundary (bnds) feature although recrystallization is happening. For clarification, Fig.1 shows new grains recrystallized at abrupt locations of the polycrystal, and Fig.2 demonstrates the unchanged grain boundaries (bnds) after recrystallization.  
![grains](https://github.com/idaholab/moose/assets/141943199/96dd1137-aa54-4816-abad-f121ffa3be71)
![boundaries](https://github.com/idaholab/moose/assets/141943199/7e1be5be-cb02-49fa-902c-5bb9349d994b)

Generally speaking, new recrystallized grains should have generated new boundaries (bnds) with the mother grains from the grain-growth model. However, it is not happening. I exactly don't know what is missing in the recrystallization model. Maybe some sort of coupling? For your kind information, the recrystallization model employs

[Kernels]
  [./c_force]
    type = DiscreteNucleationForce
    variable = 'gr10'
    map = map
    no_nucleus_value = -1
    nucleus_value = 2
  [../]
  [./c_react]
    type = Reaction
    variable = 'gr10'
  [../]
[]

with userobjects DiscreteNucleationInserter and DiscreteNucleationMap. I believe, @permcody may have experience with that.

I will appreciate if you can manage time to reply and help me out. Best regards.

**Comment by permcody** [comment]
What does your "GrainTracker" block look like? Do you have one or more reserve OPs set? Also, what is your `reserve_op_threshold` if you explicitly set it? The default for that value is 0.95, which is pretty high.

The GrainTracker will ignore all the noise on the reserve OPs until it "sees" a value that exceeds the `reserve_op_threshold`. At that point in time, it'll print diagnoistic information to the console and remap that value to one of the "regular" OPs for tracking in the simulation. Those recrystalizing grains won't have any impact on your simulation until that happens. In your first image, what are the values of those grains? Perhaps they haven't reached the threshold yet? You can always adjust that threshold value down if you want those grains to been "seen" a little quicker in your simulation.

**Reply by mdh23rct** [reply]
@permcody the graintracker looks like:
  [./grain_tracker]
    type = GrainTracker
    threshold = 0.2
    connecting_threshold = 0.08
    compute_var_to_feature_map = true
    flood_entity_type = elemental
    execute_on = ' initial timestep_begin'
    reserve_op = 1
    reserve_op_threshold = 0.95
    remap_grains = true
    outputs = none
  [../]

reserve_op_threshold = 0.95 is not giving any problem. I am having only one reserved OP. Meanwhile, I have worked on it a bit.
Currently, I am seeing new grain nucleation in the model. New grains are having new boundaries with the mother grains as well. Please have a look at: 

![Fig 1-grains-1](https://github.com/idaholab/moose/assets/141943199/adfd492b-8bca-493b-9dd2-93771b56b4b5)
![Fig 2-bnds-3](https://github.com/idaholab/moose/assets/141943199/c516b818-53b9-453e-8f35-ef702dcab308)

Yet, another problem is apparent. If you look closely to the new grains (bnds auxvariable), no clear boundaries exist between the new grains. 
1. Would you be kind to let me know what could be the reason for not having clear boundaries between new grains?
2. Userobject: DiscreteNucleationInserter allows only circular grain nucleation. Is there any other kernel or any other way to introduce grain nuclei of different shapes?
3. AuxKernel DiscreteNucleationAux works only on elemental auxiliary variable (monomial one). However, in my model, grain boundary variable (bnds) was a Lagrange one. So I changed the bnds type to monomial. Now, when I run the input file, an error appears "AuxVariable with name 'bnds' already exists but is of a differing type!". Please be aware that I have:

[Modules]
  [PhaseField]
    [GrainGrowth]
    []
  []
[]

[ICs]
[./bnds]
    type = BndsCalcIC
    variable = bnds
  [../]
[]

[AuxVariables]
 [./bnds]
    order = FIRST
    family = MONOMIAL
  [../]
[]

[AuxKernels]
[./bnds_aux]
    type = BndsCalcAux
    variable = bnds
    execute_on = TIMESTEP_END
  [../]
  [./bnds_auxx]
    type = DiscreteNucleationAux
    map = map
    variable = bnds
    no_nucleus_value = -1
    nucleus_value = 2
    execute_on = TIMESTEP_END
  [../]
[]

and have declared the auxiliary variable bnds just at one place. No duplicate declaration exists. What could be the reason of these error?

I would appreciate if you could manage time to reply.

Best regards.
